src/s3/commands/createBucket.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { getLogger } from '../../shared/logger'
+import { S3Node } from '../explorer/s3Nodes'
+import { DefaultCommands, Commands } from '../../shared/vscode/commands'
+import { DefaultWindow, Window } from '../../shared/vscode/window'
+import * as telemetry from '../../shared/telemetry/telemetry'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+
+/**
+ * Creates a bucket in the s3 region represented by the given node.
+ *
+ * Prompts the user for the bucket name, applying basic validations.
+ * Creates the bucket.
+ * Refreshes the node.
+ */
+export async function createBucketCommand(
+    node: S3Node,
+    window: Window = new DefaultWindow(),
+    commands: Commands = new DefaultCommands()
+): Promise<void> {
+    getLogger().debug(`CreateBucket called for ${node}`)
+
+    const bucketName = await window.inputBox.show({
+        prompt: localize('AWS.s3.createBucket.prompt', 'Create Bucket'),
+        placeHolder: localize('AWS.s3.createBucket.placeHolder', 'Bucket Name'),
+        validateInput: validateBucketName,
+    })
+
+    if (!bucketName) {
+        getLogger().info('CreateBucket cancelled')
+        telemetry.recordS3CreateBucket({ result: 'Cancelled' })
+        return
+    }
+
+    getLogger().info(`Creating bucket ${bucketName}`)
+    try {
+        const bucket = await node.createBucket({ bucketName })
+
+        getLogger().info(`Successfully created bucket ${bucket}`)
+        telemetry.recordS3CreateBucket({ result: 'Succeeded' })
+    } catch (e) {
+        getLogger().error(`Failed to create bucket ${bucketName}`, e)
+        window.message.showError(
+            localize('AWS.s3.createBucket.error.general', 'Failed to create bucket {0}', bucketName)
```

Comment:
Agree. I actually had the error message in the box but removed it when I noticed none of the other errors were doing that.

Given the constraints of error messages (no formatting, limited space, etc), it feels like the best experience might involve showing a basic message and providing a `View Logs...` button to reveal the error message & stacktrace in the output channel. Adding that behavior.

(There's an example of this being done already, but it opens the actual log file rather than the output channel.)